java/generate-offline-map/src/main/java/com/esri/arcgisruntime/generateofflinemap/MainActivity.java

```diff
@@ -0,0 +1,246 @@
+
+      // specify the extent, min scale, and max scale as parameters
+      double minScale = mMapView.getMapScale();
+      double maxScale = mMapView.getMap().getMaxScale();
+      // minScale must always be larger than maxScale
+      if (minScale <= maxScale) {
+        minScale = maxScale + 1;
+      }
+      GenerateOfflineMapParameters generateOfflineMapParameters = new GenerateOfflineMapParameters(
+          mDownloadArea.getGeometry(), minScale,
+          maxScale);
+
+      // create an offline map offlineMapTask with the map
+      OfflineMapTask offlineMapTask = new OfflineMapTask(mMapView.getMap());
+
+      // create an offline map job with the download directory path and parameters and start the job
+      GenerateOfflineMapJob job = offlineMapTask.generateOfflineMap(generateOfflineMapParameters, tempDirectoryPath);
+
+      // replace the current map with the result offline map when the job finishes
+      job.addJobDoneListener(() -> {
+        if (job.getStatus() == Job.Status.SUCCEEDED) {
+          GenerateOfflineMapResult result = job.getResult();
+          mMapView.setMap(result.getOfflineMap());
+          mGraphicsOverlay.getGraphics().clear();
+          mTakeMapOfflineButton.setEnabled(false);
+          Toast.makeText(this, "Now displaying offline map.", Toast.LENGTH_LONG).show();
+        } else {
+          String error = "Error in generate offline map job: " + job.getError().getAdditionalMessage();
+          Toast.makeText(this, error, Toast.LENGTH_LONG).show();
+          Log.e(TAG, error);
+        }
+        progressDialog.dismiss();
+      });
+
+      // show the job's progress with the progress dialog
+      job.addProgressChangedListener(() -> progressDialog.setProgress(job.getProgress()));
+
+      // start the job
+      job.start();
+    });
+
+  }
+
+  /**
+   * Handle the permissions request response.
+   */
+  @Override
+  public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+    if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+      generateOfflineMap();
+      Log.d(TAG, "permission granted");
+    } else {
+      // report to user that permission was denied
+      Toast.makeText(this, getString(R.string.offline_map_write_permission_denied), Toast.LENGTH_SHORT).show();
+    }
+  }
+
+  @Override
+  protected void onPause() {
+    mMapView.pause();
+    super.onPause();
+  }
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+    mMapView.resume();
+  }
+
+  @Override
+  protected void onDestroy() {
+    mMapView.dispose();
+    super.onDestroy();
+  }
+
+  /**
+   * Recursively deletes all files in the given directory.
+   *
+   * @param file to delete
+   */
+  private static void deleteDirectory(File file) {
+    if (file.isDirectory())
+      for (File subFile : file.listFiles()) {
+        deleteDirectory(subFile);
+      }
+    file.delete();
```

Comment:
It would be useful to know if i file was not deleted for any reason. Maybe add an Logcat error.